### PR TITLE
Prevent cascade deletion of pages when a ContentType is deleted

### LINKED ISSUE #2024: Change on_delete action on Page.content_type
Currently, the ``on_delete`` action on ``Page.content_type`` is set to Django's default ``models.CASCADE``. This could cause a bunch of pages to be "badly" deleted if their specific model is removed, causing their children to be orphaned and their parent's ``numchild`` to not be updated.

We've discussed this and decided that the ``on_delete`` action chould be changed to ``models.SET`` setting the ``content_type`` field to the ``ContentType`` representing the standard ``Page`` model.

https://docs.djangoproject.com/en/1.9/ref/models/fields/#django.db.models.SET

### PR DESCRIPTION
**Pull request by kaedroho**
Fixes #2024

### CODE REVIEW

**Comment by kaedroho** [comment]
Drone seems to be dead. ``flake8 wagtail`` passes locally

File: wagtail/wagtailcore/models.py [hunk 1]
```diff
@@ -207,6 +207,14 @@ def get_page_models():
     return PAGE_MODEL_CLASSES
 
 
+def get_default_page_content_type():
+    """
+    Returns the content type to use as a default for pages which their
```
Comment (tomdyson) [inline wagtail/wagtailcore/models.py]: I don't understand this sentence - should it be 'when their content type'? or 'whose content type'?

**Comment by tomdyson** [comment]
@timheap could you have a look at this, please?

File: wagtail/wagtailcore/models.py (hunk 1 above)
Comment (timheap) [inline wagtail/wagtailcore/models.py]: I agree, this wording is awkward. Consider something like:

> Get the content type to use as a default when the content type for a specific Page type is deleted

**Comment by timheap** [comment]
Looks good, besides from the docstring on `get_default_page_content_type()`!

**Comment by tomdyson** [comment]
Thanks a lot, @kaedroho and @timheap

**Comment by gasman** [comment]
Merged in 078491b6e068cb5bf7ba9f190cfc3ac085548c28

**Pull request closed without merging**